[fontconfig] Build error

**Host Environment**
- OS: Windows 10
- Compiler: gcc version 11.2.0 (x86_64-posix-seh-rev3, Built by MinGW-W64 project)
-  vcpkg-tool version: 2022-05-05-67e17c1782801cf481be9ac0b3765dff3e4bdeb8
-  vcpkg-scripts version: 314018fef 2022-05-20

**To Reproduce**
Steps to reproduce the behavior:
`vcpkg install fontconfig:x64-mingw-static`

**Failure logs**
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: D:/vcpkg/downloads/tools/python/python-3.10.2-x64/python.exe D:/vcpkg/installed/x64-windows/tools/meson/meson.py -Ddoc=disabled -Dcache-build=disabled -Dtests=disabled --buildtype plain --backend ninja --wrap-mode nodownload --native D:/vcpkg/buildtrees/fontconfig/meson-native-x64-mingw-static.log --default-library static --libdir lib --native D:/vcpkg/buildtrees/fontconfig/meson-native-x64-mingw-static-debug.log -Ddebug=true --prefix D:/vcpkg/packages/fontconfig_x64-mingw-static/debug --includedir ../include -Dcmake_prefix_path=['D:/vcpkg/installed/x64-mingw-static/debug','D:/vcpkg/installed/x64-mingw-static','D:/vcpkg/installed/x64-mingw-static/share'] D:/vcpkg/buildtrees/fontconfig/src/2.14.0-523cc719dc.clean
    Working Directory: D:/vcpkg/buildtrees/fontconfig/x64-mingw-static-dbg
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\fontconfig\config-x64-mingw-static-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:470 (vcpkg_execute_required_process)
  ports/fontconfig/portfile.cmake:19 (vcpkg_configure_meson)
  scripts/ports.cmake:146 (include)
```

config-x64-mingw-static-dbg-out.log
```
WARNING: Recommend using either -Dbuildtype or -Doptimization + -Ddebug. Using both is redundant since they override each other. See: https://mesonbuild.com/Builtin-options.html#build-type-options
The Meson build system
Version: 0.60.2
Source dir: D:\vcpkg\buildtrees\fontconfig\src\2.14.0-523cc719dc.clean
Build dir: D:\vcpkg\buildtrees\fontconfig\x64-mingw-static-dbg
Build type: native build
Project name: fontconfig
Project version: 2.14.0
C compiler for the host machine: D:/Qt/Tools/mingw1120_64/bin/x86_64-w64-mingw32-gcc.exe (gcc 11.2.0 "x86_64-w64-mingw32-gcc.exe (x86_64-posix-seh-rev3, Built by MinGW-W64 project) 11.2.0")
C linker for the host machine: D:/Qt/Tools/mingw1120_64/bin/x86_64-w64-mingw32-gcc.exe ld.bfd 2.37
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: D:/vcpkg/downloads/tools/msys2/9a1ec3f33446b195/mingw32/bin/pkg-config.exe (0.29.2)
Run-time dependency freetype2 found: YES 24.1.18
Run-time dependency expat found: YES 2.4.8
Program python found: YES (D:/vcpkg/downloads/tools/python/python-3.10.2-x64/python.exe)
WARNING: Broken python installation detected. Python files installed by Meson might not be found by python interpreter.
 This warning can be avoided by setting "python.platlibdir" option.
WARNING: Broken python installation detected. Python files installed by Meson might not be found by python interpreter.
 This warning can be avoided by setting "python.purelibdir" option.
Has header "dirent.h" : YES 
Has header "fcntl.h" : YES 
Has header "stdlib.h" : YES 
Has header "string.h" : YES 
Has header "unistd.h" : YES 
Has header "sys/statvfs.h" : NO 
Has header "sys/vfs.h" : NO 
Has header "sys/statfs.h" : NO 
Has header "sys/param.h" : YES 
Has header "sys/mount.h" : NO 
Checking for function "link" : NO 
Checking for function "mkstemp" : YES 
Checking for function "mkostemp" : NO 
Checking for function "_mktemp_s" : YES 
Checking for function "mkdtemp" : NO 
Checking for function "getopt" : YES 
Checking for function "getopt_long" : YES 
Checking for function "getprogname" : NO 
Checking for function "getexecname" : NO 
Checking for function "rand" : YES 
Checking for function "random" : NO 
Checking for function "lrand48" : NO 
Checking for function "random_r" : NO 
Checking for function "rand_r" : NO 
Checking for function "readlink" : NO 
Checking for function "fstatvfs" : NO 
Checking for function "fstatfs" : NO 
Checking for function "lstat" : NO 
Checking for function "mmap" : NO 
Checking for function "vprintf" : YES 
Checking for function "FT_Get_BDF_Property" with dependency freetype2: YES 
Checking for function "FT_Get_PS_Font_Info" with dependency freetype2: YES 
Checking for function "FT_Has_PS_Glyph_Names" with dependency freetype2: YES 
Checking for function "FT_Get_X11_Font_Format" with dependency freetype2: YES 
Checking for function "FT_Done_MM_Var" with dependency freetype2: YES 
Header <fcntl.h> has symbol "posix_fadvise" : NO 
Checking whether type "struct statvfs" has member "f_basetype" : NO 
Checking whether type "struct statvfs" has member "f_fstypename" : NO 
Checking whether type "struct statfs" has member "f_flags" : NO 
Checking whether type "struct statfs" has member "f_fstypename" : NO 
Checking whether type "struct dirent" has member "d_type" : NO 
Checking for size of "void *" : 8
Checking for alignment of "void *" : 8
Checking for alignment of "double" : 8
Checking if "stdatomic.h atomics" : links: YES 
Checking if "Intel atomics" : links: YES 
Checking if "Solaris atomic ops" : links: NO 

..\src\2.14.0-523cc719dc.clean\meson.build:208:2: ERROR: C shared or static library 'getopt' not found

A full log can be found at D:\vcpkg\buildtrees\fontconfig\x64-mingw-static-dbg\meson-logs\meson-log.txt
```

**Additional context**
I try to install getopt:x64-mingw-static, but I find that no `.a` library file was generated.
In contrast, installing `getopt-win32:x64-windows-static` can provide `getopt.lib`.
I'm sorry I don't know if fontconfig can be compiled as a static library under mingw on Windows.

--- a/ports/fontconfig/fix-mingw-gperf-fallback.patch
+++ b/ports/fontconfig/fix-mingw-gperf-fallback.patch
@@ -1,0 +1,13 @@
+diff --git a/meson.build b/meson.build
+index e5096bf..51c289f 100644
+--- a/meson.build
++++ b/meson.build
+@@ -268,7 +268,7 @@ sh = find_program('sh', required : false)
+ 
+ if not sh.found() # host_machine.system() == 'windows' or not sh.found()
+   # TODO: This is not always correct
+-  if cc.get_id() == 'msvc'
++  if host_machine.system() == 'windows'
+     gperf_len_type = 'size_t'
+   else
+     gperf_len_type = 'unsigned'

--- a/ports/fontconfig/libgetopt.patch
+++ b/ports/fontconfig/libgetopt.patch
@@ -120,7 +120,7 @@ index f616600..6d82a16 100644
 +if host_machine.system() == 'windows'
 +  conf.set('HAVE_GETOPT', 1)
 +  conf.set('HAVE_GETOPT_LONG', 1)
-+  getopt_dep = cc.find_library('getopt', required: true)
++  getopt_dep = cc.find_library('getopt', required: false)
 +else
 +  getopt_dep = dependency('', required: false)
 +endif

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -10,6 +10,7 @@ vcpkg_from_gitlab(
     PATCHES
         no-etc-symlinks.patch
         libgetopt.patch
+        fix-mingw-gperf-fallback.patch
 )
 
 vcpkg_add_to_path(PREPEND "${CURRENT_HOST_INSTALLED_DIR}/tools/gperf")

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "fontconfig",
   "version": "2.14.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Library for configuring and customizing font access.",
   "homepage": "https://www.freedesktop.org/wiki/Software/fontconfig",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2314,7 +2314,7 @@
     },
     "fontconfig": {
       "baseline": "2.14.0",
-      "port-version": 2
+      "port-version": 3
     },
     "foonathan-memory": {
       "baseline": "2019-07-21",

--- a/versions/f-/fontconfig.json
+++ b/versions/f-/fontconfig.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "edebefba0511e2f8ee4018707fe611d0035c4dd4",
+      "version": "2.14.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "5c02d2ba027c9f390729a7e0899e63c853450dfb",
       "version": "2.14.0",
       "port-version": 2

